cassiopeia/tools/parameter_estimators.py

```diff
@@ -0,0 +1,426 @@
+    in depth or time. The average depth/lineage time of the tree is used as a
+    proxy for the depth/total time when the tree is not ultrametric.
+
+    In the inference, we need to consider whether to assume an implicit root,
+    specified by `assume_root_implicit_branch`. In the case where the tree does
+    not have a single leading edge from the root representing the progenitor
+    cell before cell division begins, this additional edge is added to the
+    total time in calculating the estimate if `assume_root_implicit_branch` is
+    True.
+
+    Args:
+        tree: The CassiopeiaTree specifying the tree and the character matrix
+        continuous: Whether to calculate a continuous mutation rate,
+            accounting for branch lengths. Otherwise, calculates a
+            discrete mutation rate using the node depths
+        assume_root_implicit_branch: Whether to assume that there is an
+            implicit branch leading from the root, if it doesn't exist
+        layer: Layer to use for character matrix. If this is None,
+            then the current `character_matrix` variable will be used.
+
+    Returns:
+        The estimated mutation rate
+
+    Raises:
+        ParameterEstimateError if the `mutation_proportion` parameter is not
+            between 0 and 1
+    """
+    if "mutated_proportion" not in tree.parameters:
+        mutation_proportion = get_proportion_of_mutation(tree, layer)
+    else:
+        mutation_proportion = tree.parameters["mutated_proportion"]
+    if mutation_proportion < 0 or mutation_proportion > 1:
+        raise ParameterEstimateError(
+            "Mutation proportion must be between 0 and 1."
+        )
+    if not continuous:
+        mean_depth = tree.get_mean_depth_of_tree()
+        # We account for the added depth of the implicit branch leading
+        # from the root, if it is to be added
+        if assume_root_implicit_branch and len(tree.children(tree.root)) != 1:
+            mean_depth += 1
+        mutation_rate = 1 - (1 - mutation_proportion) ** (1 / mean_depth)
+    else:
+        times = tree.get_times()
+        mean_time = np.mean([times[l] for l in tree.leaves])
+        if assume_root_implicit_branch and len(tree.children(tree.root)) != 1:
+            mean_time += np.mean(
+                [tree.get_branch_length(u, v) for u, v in tree.edges]
+            )
+        mutation_rate = -np.log(1 - mutation_proportion) / mean_time
+    return mutation_rate
+
+
+def estimate_missing_data_rates(
+    tree: CassiopeiaTree,
+    continuous: bool = True,
+    assume_root_implicit_branch: bool = True,
+    stochastic_missing_probability=None,
+    heritable_missing_rate=None,
+    layer: Optional[str] = None,
+) -> Tuple[float, float]:
+    """
+    Estimates both missing data parameters given one of the two from a tree.
+
+    The stochastic missing probability is the probability that any given
+    cell/character pair acquires stochastic missing data in the character
```

Comment:
Can you briefly describe what the mechanism is behind "stochastic" missing data?